crates/re_space_view_spatial/src/scene/parts/points2d.rs

```diff
@@ -157,67 +161,68 @@ impl Points2DPart {
                     }
                 }
             }
-        }
+        };
+
+        load_keypoint_connections(ent_context, ent_path, keypoints);
 
-        scene.load_keypoint_connections(ent_path, keypoints, &annotations);
+        {
+            re_tracing::profile_scope!("points2d.bounding_box");
+            self.data.bounding_box = self.data.bounding_box.union(
+                macaw::BoundingBox::from_points(
+                    ent_view
+                        .iter_primary()?
+                        .filter_map(|pt| pt.map(|pt| glam::vec3(pt.x, pt.y, 0.0))),
+                )
+                .transform_affine3(&ent_context.world_from_obj),
+            );
+        }
 
         Ok(())
     }
 }
 
-impl ScenePart for Points2DPart {
-    fn load(
-        &self,
-        scene: &mut SceneSpatial,
+impl ScenePartImpl for Points2DPart {
+    type SpaceViewState = SpatialSpaceViewState;
+    type SceneContext = SpatialSceneContext;
```

Comment:
haven't found a good way to avoid typing that out every time. `trait` implementing `trait` is really awful unless all traits involved are in the same crate (orphan rule kicks in otherwise along the way)

Reply:
I think you could get rid of 1 line by changing to:
```
trait ScenePartImpl {
  type SpaceViewClass: SpaceViewClassImpl;
  ...

impl ScenePartImpl for Points2DPart {
  type SpaceViewClass = SpacialSpaceView;
```

And then you should be able to refer to: `SpaceViewClass::SpaceViewState`.

This feels logically appropraite since it's basically a way of specifying which SpaceViewClass a given ScenePart belongs to.

Reply:
I like the idea but rust doesn't allow me to do this it seems. I tried
`type SpaceViewClass: SpaceViewClassImpl;`

but then using `Self::SpaceViewClass::SpaceViewState` gives me

```
 1  error[E0223]: ambiguous associated type
   --> crates/re_viewer_context/src/space_view/scene_part_impl.rs:21:28
    |
 21 |         space_view_state: &Self::SpaceViewClass::SpaceViewState,
    |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
 help: if there were a trait named `Example` with associated type `SpaceViewState` implemented for `<Self as space_view::scene_part_impl::Sce
 nePartImpl>::SpaceViewClass`, you could use the fully-qualified path
    |
 21 |         space_view_state: &<<Self as space_view::scene_part_impl::ScenePartImpl>::SpaceViewClass as Example>::SpaceViewState,
    |                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
    
Just to be sure I tried using different names for the various `type` (since I re-use some monikers there atm), but that didn't help

Reply:
Hmmm interesting so this is a problem with associated types only

Reply:
But the problem is then that neither
```impl<C: SpaceViewClassImpl, T: ScenePartImpl<C> + 'static> ScenePart for T```
nor
```impl<T: ScenePartImpl + 'static> ScenePart for T {```
is allowed. But maybe the whole thing would work out with your typed scene proposal after all